app/routes/index.ts

```diff
@@ -0,0 +1,12 @@
+import Store from '@ember-data/store';
+import Route from '@ember/routing/route';
+import { service } from '@ember/service';
+import RentalModel from 'super-rentals/models/rental';
+
+export default class IndexRoute extends Route {
+  @service declare store: Store;
+
+  async model(): Promise<Array<RentalModel>> {
+    return (await this.store.findAll('rental')).toArray();
```

Comment:
I linked to this comment in the blog post, but I won't be changing the code since I don't want to stray too far from the behavior in the original Super Rentals tutorial. With that said, should we consider changing the behavior there? I was aware of the differences you mentioned above and I've read the docs pretty thoroughly.

Also, if anyone is following along, the code that actually works is:

```ts
  async model(): Promise<ArrayProxy<RentalModel>> {
    return this.store.query('rental', {});
  }
```

Without the `{}`, the code doesn't pass type-checking and throws this error in runtime: `Error: Assertion Failed: You need to pass a query hash to the store's query method`

Reply:
> With that said, should we consider changing the behavior there? I was unaware of the differences you mentioned above and I've read the docs pretty thoroughly.

I definitely think it should be considered. `findAll` is one of the "bad parts" of ember-data I often find myself needing to steer folks away from due to how complicated the matrix can be for whether it fires a request or not.

The downside of `query` is that repeat calls will always hit network unless you opt to put your own "query cache" in front of it. This is something we're likely going to change this year in ember-data via RFC, as most every app I've seen has written the same "query cache" logic to account for this.